src/bitcoin-cli.cpp

```diff
@@ -404,7 +404,7 @@ static UniValue CallRPC(BaseRequestHandler *rh, const std::string& strMethod, co
     // Parse reply
     UniValue valReply(UniValue::VSTR);
     if (!valReply.read(response.body))
-        throw std::runtime_error("couldn't parse reply from server");
+        throw std::runtime_error(strprintf("server response: %s", response.body));
```

Comment:
I think it's strange that it ends up here. It's trying to parse an error message as JSON! Isn't there some error code (either HTTP or JSON-RPC) that could be checked for instead? If not, that seems like a bug.

Reply:
It does seem wrong that my proposed fix retains the practice of trying to decode a non-JSON string as JSON, and depending on that to fail. Yes, perhaps this is a bug in the server -- should it be returning JSON in `body` here rather than a plain string?  But as I said at the end of the comment above, I was hesitant to change the server because I'm worried that there may be other RPC clients besides `bitcoin-cli` that depend on the current (plain string) response. What do you think? If the fix is better in the server, I can do that. 

Just for further information, the response returned in this work queue depth exceeded is (status 500 is [`HTTP_INTERNAL_SERVER_ERROR`](https://github.com/bitcoin/bitcoin/blob/master/src/rpc/protocol.h#L18)):
```
(gdb) f 0
Breakpoint 1, CallRPC (rh=0x555555845e90, strMethod="getblockhash", args=std::vector of length 1, capacity 2 = {...})
    at bitcoin-cli.cpp:406
406	    if (!valReply.read(response.body))
(gdb) p response
$2 = {
  status = 500, 
  error = -1, 
  body = "Work queue depth exceeded"
}
(gdb) 
```
Here's a successful reply:
```
(gdb) p response
$6 = {
  status = 200, 
  error = -1, 
  body = "{\"result\":\"000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f\",\"error\":null,\"id\":1}\n"
}
```
Here's the response when requesting a nonsense method:
```
(gdb) p response
$2 = {
  status = 404, 
  error = -1, 
  body = "{\"result\":null,\"error\":{\"code\":-32601,\"message\":\"Method not found\"},\"id\":1}\n"
}
```
As part of this PR, I could also systematically search for and fix any other places where `body` is returned as plain string (not JSON); so far I see only one:
```
req->WriteReply(HTTP_BAD_METHOD, "JSONRPC server handles only POST requests")
```